environment/trading_env.py

```diff
@@ -137,18 +146,48 @@ def step(
         info = {}
         return observation, reward, terminated, truncated, info
 
-    def _update_positions(self, new_positions):
+    def _get_relative_close_prices(self):
         """
-        1. p_(t-1) = self._positions
-        2. p_t' = (today_close * p_(t-1)) / (today_close.dot(p_(t-1))) (L1 norm of today_close * self._positions)
-        3. p_t = new_positions
-        4. transaction_cost = today_close.dot(abs(p_t' - p_t)) * t_cost_percentage
-        5. capital = today_close.dot(p_t) - transaction_cost
+        If an asset changes price from $100 to $120, this would be a 1.2 relative price change.
+        First element is the relative change in cash, which is generally 1.
         """
-
         yesterday_close, today_close = self.get_relevant_close_prices()
-        old_positions = self._positions
-        market_adjusted_positions = today_close * old_positions
+        rel = today_close / yesterday_close
+
+        CASH_CHANGE = 0
+        rel = np.append([1 + CASH_CHANGE], rel)
+
+        return rel
+
+    def _get_market_adjusted_positions(self) -> np.array:
+        rel_prices = self._get_relative_close_prices()
+        return positions * rel_prices / positions.dot(rel_prices)
```

Comment:
Positions doesn't appear to be defined, did you mean something along the lines of `self._positions`?